diff --git a/dune/grid-glue/merging/contactmerge.hh b/dune/grid-glue/merging/contactmerge.hh
index 215e20a..b500cea 100644
--- a/dune/grid-glue/merging/contactmerge.hh
+++ b/dune/grid-glue/merging/contactmerge.hh
@@ -8,6 +8,9 @@
 #ifndef DUNE_GRIDGLUE_MERGING_CONTACTMERGE_HH
 #define DUNE_GRIDGLUE_MERGING_CONTACTMERGE_HH

+#ifndef M_PI
+#define M_PI           3.14159265358979323846
+#endif

 #include <iostream>
 #include <fstream>
